packages/definitions-parser/src/packages.ts

```diff
@@ -448,12 +508,30 @@ export class TypingsData extends PackageBase {
     return this.data.typesVersions;
   }
 
-  get files(): readonly string[] {
-    return this.data.files;
+  private typesVersionsFiles: readonly FilesForSingleTypeScriptVersion[] | undefined;
+  tryGetFiles(): readonly string[] | { errors: string[] } {
+    if (!this.typesVersionsFiles) {
+      const files = getFiles(this.dt, this, this.moduleResolutionHost);
+      if ("errors" in files) {
+        return files;
+      }
+      this.typesVersionsFiles = files;
+    }
+    return this.typesVersionsFiles.flatMap((v) => v.declFiles);
   }
-  get dtsFiles(): readonly string[] {
-    return this.data.files.filter((f) => f.endsWith(".d.ts") || f.endsWith(".d.mts") || f.endsWith(".d.cts"));
+
+  getFiles(): readonly string[] {
+    const files = this.tryGetFiles();
+    if ("errors" in files) {
+      throw new Error(`Errors while reading package ${this.name}: ${files.errors.join("\n")}`);
```

Comment:
did this throw before? I think that it read from the (virtual) filesystem -- so what kind of errors actually get thrown here?

Mainly asking because I thought tryGetFiles would be the more-used method, but getFiles turns out to be.

Reply:
These errors would have been thrown way earlier, during the parsing of all definitions. In the end, I think changing the error handling here wasn’t significant. I did it this way because I initially kept this as a `get files()` accessor, which I didn’t want to throw on access. But I changed it to a method call to ensure you couldn’t generate errors without noticing. So I think I can simplify this a bit.

Reply:
> what kind of errors actually get thrown here

This is doing way more than reading. This is the process where we create TS source files and perform module resolution on imports/references so we can see what files are used/unused and check that packages aren’t using relative imports of other packages.